scanner/src/main/kotlin/storages/PostgresStorage.kt

```diff
@@ -193,6 +201,61 @@ class PostgresStorage(
         }
     }
 
+    override fun readInternal(
+        packages: List<Package>,
+        scannerCriteria: ScannerCriteria
+    ): Result<Map<Identifier, List<ScanResult>>> {
+        if (packages.isEmpty()) return Success(emptyMap())
+
+        val minVersionArray = with(scannerCriteria.minVersion) { intArrayOf(major, minor, patch) }
+        val maxVersionArray = with(scannerCriteria.maxVersion) { intArrayOf(major, minor, patch) }
+
+        @Suppress("TooGenericExceptionCaught")
+        return try {
+            val scanResults = runBlocking(Dispatchers.IO) {
+                packages.chunked(max(packages.size / 50, 1)).map { chunk ->
```

Comment:
This dynamic definition of the chunk size I find a bit strange. I would assume that for smaller projects it is more efficient to run few queries in parallel that load more packages than running many queries that only deal with few packages. So a 1 as lower bound for the chunk size seems to be too small.

For huge projects OTOH, you may want to restrict the chunk size to keep the memory consumption in a certain limit. (This is probably not an issue for now as all scan results are kept in memory anyway, but maybe later if some kind of streaming was used.)

So, should the chunk size be made configurable? There could be even options for a maximum and minimum chunk size; the chunk size could then still be calculated based on the number of packages, but then aligned with these bounds.

Reply:
> I would assume that for smaller projects it is more efficient to run few queries in parallel that load more packages than running many queries that only deal with few packages.

Actually, this depends a lot on the bandwidth of the connection, when testing with a VPN connection many small queries were faster than one large query, the same test with a good connection does not show any relevant difference, but I would expect it causes higher load on the DB. But anyway, for such small projects downloading the results takes usually only a few seconds.

> So, should the chunk size be made configurable?

Yes, I wanted to do this but then realized that this code will be rewritten anyway next week when implementing https://github.com/oss-review-toolkit/ort/issues/2668, so I decided to do this first, because I don't know yet what these changes will look like exactly.

As all my tests have shown at least equal but usually better performance than with the previous code, I would suggest to first implement #2668 and then look into performance optimization and making it configurable.

Reply:
Ok, makes sense.